http-server/src/main/java/io/micronaut/http/server/cors/CorsFilter.java

```diff
@@ -132,27 +132,51 @@ protected boolean shouldDenyToPreventDriveByLocalhostAttack(@NonNull CorsOriginC
         if (origin == null) {
             return false;
         }
-        if (origin.startsWith(LOCALHOST)) {
+        if (isOriginLocal(origin)) {
             return false;
         }
         String host = httpHostResolver.resolve(request);
-        return isAny(corsOriginConfiguration.getAllowedOrigins()) && host.startsWith(LOCALHOST);
-
+        return isAny(corsOriginConfiguration.getAllowedOrigins()) && isHostLocal(host);
     }
 
     /**
      *
      * @param origin HTTP Header {@link HttpHeaders#ORIGIN} value.
      * @param request HTTP Request
-     * @return {@literal true} if the resolved host starts with {@literal http://localhost} and origin does not start with localhost deny it.
+     * @return {@literal true} if the resolved host is localhost or 127.0.0.1 and origin is not one of these then deny it.
      */
     protected boolean shouldDenyToPreventDriveByLocalhostAttack(@NonNull String origin,
                                                                 @NonNull HttpRequest<?> request) {
         if (httpHostResolver == null) {
             return false;
         }
         String host = httpHostResolver.resolve(request);
-        return !origin.startsWith(LOCALHOST) && host.startsWith(LOCALHOST);
+        return !isOriginLocal(origin) && isHostLocal(host);
+    }
+
+    /*
+     * We only need to check host for starting with "localhost" "127." (as there are multiple loopback addresses on linux)
+     *
+     * This is fine for host, as the request had to get here.
+     */
+    private boolean isHostLocal(@NonNull String hostString) {
+        return hostString.startsWith("http://localhost")
+            || hostString.startsWith("https://localhost")
+            || hostString.startsWith("http://127.")
+            || hostString.startsWith("https://127.")
+            || hostString.startsWith("ws://localhost")
+            || hostString.startsWith("wss://localhost")
+            || hostString.startsWith("ws://127.")
+            || hostString.startsWith("wss://127.");
```

Comment:
can we improve the performance of this with a check that the first char is `w` or `h`?

Reply:
Added a pre-check for a host of length 0 and that the first char must be h or w